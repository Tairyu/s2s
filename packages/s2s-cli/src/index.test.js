// @flow
import type { Opts } from 'types'
import m from '.'

let logSpy
const watchers = []

beforeEach(() => {
  logSpy = jest.spyOn(console, 'log')
  logSpy.mockImplementation(x => x)
})

afterEach(() => {
  logSpy.mockRestore()
  for (const w of watchers) {
    w.close()
  }
})

const setup = (opts: $Shape<Opts>) => ({
  watch: 'app',
  plugins: [],
  templates: [],
  ...opts,
})

const addToWatchers = (opts: $Shape<Opts>) => () => {
  watchers.push(m(setup(opts)))
}

test('export function', () => {
  expect(typeof m).toBe('function')
})

test('throw error when opts.watch == null', () => {
  expect(() => {
    // $FlowFixMe
    m({ watch: null })
  }).toThrowErrorMatchingSnapshot()
})

test('throw error when opts.plugins !== Array', () => {
  expect(() => {
    // $FlowFixMe
    m(setup({ plugins: 'not array' }))
  }).toThrowErrorMatchingSnapshot()
})

test('throw error when opts.templates !== Array', () => {
  expect(() => {
    // $FlowFixMe
    m(setup({ templates: 'string' }))
  }).toThrowErrorMatchingSnapshot()
})

test('throw error when opts.after !== Array', () => {
  expect(() => {
    // $FlowFixMe
    m(setup({ afterHooks: 'string' }))
  }).toThrowErrorMatchingSnapshot()
})

test('not throw when no args', () => {
  expect(addToWatchers()).not.toThrow()
})

test('Displayed "start" on the Consle', () => {
  expect(addToWatchers()).not.toThrow()
  expect(logSpy.mock.calls.length).toBe(1)
  expect(logSpy.mock.calls[0][0]).toMatch('start')
})
