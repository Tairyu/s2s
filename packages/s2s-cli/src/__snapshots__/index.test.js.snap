// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throw errors opts.after !== Array 1`] = `"Expected a Array got string"`;

exports[`throw errors opts.plugins !== Array 1`] = `"Expected a Array, got string"`;

exports[`throw errors opts.templates !== Array 1`] = `"Expected a Array got string"`;

exports[`throw errors opts.watch == null 1`] = `"required watch"`;
